lightningd/channel_control.c

```diff
@@ -322,3 +326,100 @@ bool channel_tell_funding_locked(struct lightningd *ld,
 		lockin_complete(channel);
 	return true;
 }
+
+/* Check if we are the fundee of this channel, the channel
+ * funding transaction is still not yet seen onchain, and
+ * it has been too long since the channel was first opened.
+ * If so, we should forget the channel. */
+static bool
+is_fundee_should_forget(struct lightningd *ld,
+			struct channel *channel,
+			u32 block_height)
+{
+	u32 max_no_funding_tx = 2016;
+
+	/* FIXME: we should be getting max_no_funding_tx
+	 * from an lightningd option, which is why we get
+	 * it as an argument. */
+	(void) ld;
+
+	/* BOLT #2:
+	 *
+	 * A non-funding node (fundee):
+	 *   - SHOULD forget the channel if it does not see the
+	 * funding transaction after a reasonable timeout.
+	 */
+
+	/* Only applies if we are fundee. */
+	if (channel->funder == LOCAL)
+		return false;
+
+	/* Does not apply if we already saw the funding tx. */
+	/* FIXME: This is only set when the funding tx is confirmed
```

Comment:
Don't think this is a huge problem, and I think seeing it in the mempool could still hold us up indefinitely due to low fees. Happy to keep this as is. `scid` is set after the first confirmation.